components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle

```diff
@@ -55,15 +55,18 @@ void configureGradleEnterprise(gradleEnterprise) {
                 server = projectProperties.get("com.gradle.enterprise.build_validation.server")
             }
 
-            if (!server) {
+            if (!server && !gradle.startParameter.noBuildScan) {
```

Comment:
There is a handful of ways we could write these conditionals. We could use `gradle.startParameter.noBuildScan`, `gradle.startParameter.buildScan`, or `System.properties.containsKey("scan.dump")`. We could also leverage `publishAlwaysIf(...)` and remove `--no-scan` as a command line argument.

The bottom line is when running with `-x` we also need to explicitly disable Build Scan publishing in the init script. 

Let me know what you think is cleanest.

Reply:
No longer relevant as I have changed the code.